kyaml/yaml/rnode.go

```diff
@@ -756,14 +756,17 @@ func (rn *RNode) FieldRNodes() ([]*RNode, error) {
 // Field returns a fieldName, fieldValue pair for MappingNodes.
 // Returns nil for non-MappingNodes.
 func (rn *RNode) Field(field string) *MapNode {
-	if rn.YNode().Kind != yaml.MappingNode {
-		return nil
+	if yNode := rn.YNode(); yNode != nil {
+		if yNode.Kind != yaml.MappingNode {
```

Comment:
Please don't make the indent too deep here.
I think it's better to first store the value in the local variable of `yNode` and then check with `if yNode == nil` in an early-return format.